lib/puppet/provider/netapp_storage_failover/cmode.rb

```diff
@@ -0,0 +1,79 @@
+require_relative '../netapp_cmode'
+
+Puppet::Type.type(:netapp_storage_failover).provide(:cmode, :parent => Puppet::Provider::NetappCmode) do
+  @doc = "Manage storage failover. [Family: cluster]"
+
+  confine    :feature => :posix
+  defaultfor :feature => :posix
+
+  netapp_commands   :strgfailovershow  => {:api => 'cf-get-iter', :iter => true, :result_element => 'attributes-list'}
+  netapp_commands   :strgfailovermdfy  => 'cf-modify-iter'
+  mk_resource_methods
+
+  def self.instances
+    Puppet.debug("Puppet::Provider::Netapp_storage_failover.cmode: Got to self.instances")
+    strgfailoverinfos = []
+    results = strgfailovershow() || []
+    
+    results.each do |storage_failover_info|
+      sfo_options_info  = storage_failover_info.child_get('sfo-options-info')
+      options_related_info = sfo_options_info.child_get('options-related-info')
+      sfo_giveback_options_info = options_related_info.child_get('sfo-giveback-options-info')
+      giveback_options = sfo_giveback_options_info.child_get('giveback-options')
```

Comment:
`sfo_giveback_options` can be `nil` when storage failover is not enabled causing this to fail with:
```
Error: Could not run: undefined method `child_get' for nil:NilClass
/vagrant/puppetlabs-netapp/lib/puppet/provider/netapp_storage_failover/cmode.rb:22:in `block in instances'
```

Reply:
Hello Hunner,

Thank you very much for reviewing it.

But the customer ask/expectation from this type/provider, was to have a capability, to modify 3 parameters related to giveback.
1)  giveback_options_modify.child_add_string('auto-giveback-after-panic-enabled', @resource[:auto_giveback_after_panic]) unless @resource[:auto_giveback_after_panic].nil?

2) giveback_options_modify.child_add_string('auto-giveback-enabled', @resource[:auto_giveback]) unless @resource[:auto_giveback].nil?

3) giveback_options_modify.child_add_string('auto-giveback-override-vetoes-enabled', @resource[:auto_giveback_override_vetoes]) unless @resource[:auto_giveback_override_vetoes].nil?

Hence we assumed sfo_giveback_options would be not nil.

Still i will put a check and resubmit, Thanks for pointing this out. Thank you.